### PR TITLE
Feature 2290 confirm review period

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 1]
```diff
@@ -370,7 +370,7 @@ const TeamReviews = ({ onBack, periodId }) => {
 
   const handleLaunchDateChange = (val, period) => {
     const newDate = val?.$d;
-    const isoDate = newDate.toISOString() ?? null;
+    const isoDate = newDate?.toISOString() ?? null;
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:373]: When this component is initially loaded and the review period has no launch date yet, the `DatePicker` for launch date automatically opens to the calendar view and `$d` has no value. Adding the `?` here addresses that.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 2]
```diff
@@ -527,7 +527,28 @@ const TeamReviews = ({ onBack, periodId }) => {
     }
   };
 
+  const validateReviewPeriod = period => {
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:530]: This function implements review period validation that must hold in order to "REQUEST APPROVAL".

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ import { useLocation, useHistory } from 'react-router-dom';
 
 import { AddCircle, Archive, Delete, Unarchive } from '@mui/icons-material';
 import {
+  Alert,
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:17]: Using for validation error messages.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 4]
```diff
@@ -398,7 +400,7 @@ const TeamReviews = ({ onBack, periodId }) => {
 
   const handleCloseDateChange = (val, period) => {
     const newDate = val?.$d;
-    const isoDate = newDate.toISOString() ?? null;
+    const isoDate = newDate?.toISOString() ?? null;
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:403]: If no launch date was specified, the `?` here prevents an error from trying to call `toISOString`.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 5]
```diff
@@ -384,7 +386,7 @@ const TeamReviews = ({ onBack, periodId }) => {
 
   const handleSelfReviewDateChange = (val, period) => {
     const newDate = val?.$d;
-    const isoDate = newDate.toISOString() ?? null;
+    const isoDate = newDate?.toISOString() ?? null;
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:389]: If no launch date was specified, the `?` here prevents an error from trying to call `toISOString`.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 6]
```diff
@@ -689,13 +720,14 @@ const TeamReviews = ({ onBack, periodId }) => {
...
           />
-          <Button onClick={requestApproval}>
-            {period.reviewStatus === ReviewStatus.AWAITING_APPROVAL
-              ? 'Launch Review'
-              : 'Request Approval'}
-          </Button>
+          {approvalButton()}
         </div>
       )}
+      {validationMessage && (
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:726]: If there is a validation error message, display it in an `Alert`.

**Review by vhscom — APPROVED** [review]
LGTM

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 7]
```diff
@@ -627,6 +648,17 @@ const TeamReviews = ({ onBack, periodId }) => {
     );
   };
 
+  const approvalButton = () => {
```
[the remark this answers is not in the document]
Comment (jackkeller) [reply web-ui/src/components/reviews/TeamReviews.jsx:650]: Without knowing off hand how many `ReviewStatus` potentials there are could this have been a ternary?

**Review by jackkeller — APPROVED** [review]
Left a comment, not a dealbreaker.

File: web-ui/src/components/reviews/TeamReviews.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (mvolkmann) [reply web-ui/src/components/reviews/TeamReviews.jsx:650]: There are currently five `ReviewStatus` values, so I'm thinking there will be other button labels in the future.

**Pull request merged**